tests/test_model_specs.py

```diff
@@ -57,35 +57,184 @@ def test_model_specs_are_valid(self):
-                    ("Required key(s) missing from ARGS_SPEC: "
-                     f"{set(required_keys).difference(model.ARGS_SPEC)}"))
-                extra_keys = set(model.ARGS_SPEC).difference(required_keys)
+                    required_keys.issubset(model.MODEL_SPEC),
+                    ("Required key(s) missing from MODEL_SPEC: "
+                     f"{set(required_keys).difference(model.MODEL_SPEC)}"))
+                extra_keys = set(model.MODEL_SPEC).difference(required_keys)
                 if (extra_keys):
                     self.assertEqual(extra_keys, set([optional_spatial_key]))
                     self.assertTrue(
-                        set(model.ARGS_SPEC[optional_spatial_key]).issubset(
+                        set(model.MODEL_SPEC[optional_spatial_key]).issubset(
                             {'spatial_keys', 'different_projections_ok'}))
 
             # validate that each arg meets the expected pattern
             # save up errors to report at the end
-            for key, arg in model.ARGS_SPEC['args'].items():
+            for key, arg in model.MODEL_SPEC['args'].items():
                 # the top level should have 'name' and 'about' attrs
                 # but they aren't required at nested levels
-                self.validate(arg, f'{model_name}.{key}')
+                self.validate(arg, f'{model_name}.args.{key}')
 
-    def validate(self, arg, name, parent_type=None, is_pattern=False):
+            for key, spec in model.MODEL_SPEC['outputs'].items():
+                self.validate_output(spec, f'{model_name}.outputs.{key}')
+
+    def validate_output(self, spec, key, parent_type=None):
+        """
+        Recursively validate nested output specs against the output spec standard.
+
+        Args:
+            spec (dict): any nested output spec component of a MODEL_SPEC
+            key (str): key to identify the spec by in error messages
+            parent_type (str): the type of this output's parent output (or None if
+                no parent).
+
+        Returns:
+            None
+
+        Raises:
+            AssertionError if the output spec violates the standard
+        """
+        with self.subTest(output=key):
+            # if parent_type is None:  # all top-level args must have these attrs
+            #     for attr in ['about']:
+            #         self.assertIn(attr, spec)
+
+            attrs = set(spec.keys())
+
+            if 'type' in spec:
+                t = spec['type']
+            else:
+                file_extension = key.split('.')[-1]
+                if file_extension == 'tif':
+                    t = 'raster'
+                elif file_extension in {'shp', 'gpkg', 'geojson'}:
+                    t = 'vector'
+                elif file_extension == 'csv':
+                    t = 'csv'
+                elif file_extension in {'json', 'txt', 'pickle', 'db', 'zip',
+                                        'dat', 'idx'}:
```

Comment:
Look at all the formats we output!

Reply:
Would `html` fall under here as well? Does Carbon output an `html` file?

Reply:
I think we removed that and the rest of `natcap/invest/reporting/`, didn't we?

Reply:
Carbon does still produce an html file: https://github.com/natcap/invest/blob/b437cb8d0cf74f53ce98b5386eb0f2d4a26f4342/src/natcap/invest/carbon.py#L575-L640

Reply:
Ah, got it.  So it's writing its own HTML report anot using the `reporting` subpackage.